docs/src/components/slides/make-web-games/scene/LightingSizzle.tsx

```
@@ -0,0 +1,432 @@
+  // seconds, and bounces off the interior bounds.
+  type Slime = {
+    sprite: AnimatedSprite2D | null
+    light: Light2D | null
+    pos: Vector2
+    vel: Vector2
+    retarget: number
+  }
+  const slimesRef = useRef<Slime[]>([])
+  if (slimesRef.current.length !== SLIME_COUNT) {
+    slimesRef.current = Array.from({ length: SLIME_COUNT }, () => {
+      const angle = Math.random() * Math.PI * 2
```

ESLint

[error] Error: Cannot call impure function during render

`Math.random` is an impure function. Calling an impure function can produce unstable results that update unpredictably when the component happens to re-render. (https://react.dev/reference/rules/components-and-hooks-must-be-pure#components-and-hooks-must-be-idempotent).

/home/jailuser/git/docs/src/components/slides/make-web-games/scene/LightingSizzle.tsx:149:21
  147 |   if (slimesRef.current.length !== SLIME_COUNT) {
  148 |     slimesRef.current = Array.from({ length: SLIME_COUNT }, () => {
> 149 |       const angle = Math.random() * Math.PI * 2
      |                     ^^^^^^^^^^^^^ Cannot call impure function
  150 |       return {
  151 |         sprite: null,
  152 |         light: null,

(react-hooks/purity)